demos/common/python/openvino/model_zoo/model_api/models/classification.py

```diff
@@ -26,9 +27,30 @@ def __init__(self, ie, model_path, labels, ntop, resize_type='crop'):
         if self.h != self.w:
             raise RuntimeError('Model input has incorrect image shape. Must be NxN square. '
                                 'Got {} x {}.'.format(self.h, self.w))
+
         self.ntop = ntop
         self.labels = self._load_labels(labels)
-        self.output_blob_name = self._get_outputs()
+        self._check_outputs()
+
+        function = ng.function_from_cnn(self.net)
+        if 'Softmax' not in [node.get_type_name() for node in function.get_ops()]:
+            logits = function.get_output_op(0)
+            logits = logits.inputs()[0].get_source_output().get_node()
+            logits.set_friendly_name('logits')
+            probs_node = ng.softmax(logits, 1)
+        else:
+            probs_node = function.get_output_op(0)
+            probs_node = probs_node.inputs()[0].get_source_output().get_node()
+            probs_node.set_friendly_name('probs')
+
+        topk_node = ng.topk(probs_node, ntop, 1, "max", "value")
```

Comment:
I suggest keeping topk and added softmax independent. With that IE will have a chance to compute it in parallel, though I'm not sure it exploits that.

Reply:
topk reduces number of elements in tensor. Never mind then